sagemaker-debugger/tensorflow2/tensorflow2_keras_custom_container/src/tf_keras_resnet_byoc.py

```diff
@@ -0,0 +1,88 @@
+"""
+This script is a ResNet training script which uses Tensorflow's Keras interface.
+It has been orchestrated with SageMaker Debugger hooks to allow saving tensors during training.
+These hooks have been instrumented to read from json configuration that SageMaker will put in the training container.
+Configuration provided to the SageMaker python SDK when creating a job will be passed on to the hook.
+This allows you to use the same script with differing configurations across different runs.
+If you use an official SageMaker Framework container (i.e. AWS Deep Learning Container), then
+you do not have to orchestrate your script as below. Hooks will automatically be added in those environments.
+For more information, please refer to https://github.com/awslabs/sagemaker-debugger/blob/master/docs/sagemaker.md
+"""
+
+# Standard Library
+import argparse
+import random
+
+# Third Party
+import numpy as np
+import tensorflow.compat.v2 as tf
+from tensorflow.keras.applications.resnet50 import ResNet50
+from tensorflow.keras.datasets import cifar10
+from tensorflow.keras.utils import to_categorical
+import smdebug.tensorflow as smd
+
+
+def train(batch_size, epoch, model, hook):
+    (X_train, y_train), (X_valid, y_valid) = cifar10.load_data()
+
+    Y_train = to_categorical(y_train, 10)
+    Y_valid = to_categorical(y_valid, 10)
+
+    X_train = X_train.astype('float32')
+    X_valid = X_valid.astype('float32')
+
+    mean_image = np.mean(X_train, axis=0)
+    X_train -= mean_image
+    X_valid -= mean_image
+    X_train /= 128.
+    X_valid /= 128.
+
+    hook.set_mode(smd.modes.TRAIN)
```

Comment:
Explain why it is done.  
This is not mandatory. This step will help us to differentiate TRAINING mode with EVAL mode. To learn more about modes follow {Link to doc explaining modes. }

Reply:
And may be for keras, hook will automatically get the mode, right? You can right in comment that this is optional. 
Also will this line always cause tensors to be saved in TRAIN mode, even eval mode tensors would be saved in TRAIN mode?

Reply:
removed it as it is not mandatory. as you rightly pointed out, it gets set in the callbacks